### PR TITLE
Add preload scripts for Ad assets

### LINKED ISSUE #8257: [Performance] Preload Ad (GNL) domains when we know Adverts should render
**Is your feature request related to a problem? Please describe.**
GNL have produced a set of un-versioned domains for some of their Advert logic that will enable us to preload them on pages that should serve ads. 

This comes from a recommendation from multiple performance related tests that show we should preload these domains to help improve our frontend web performance

**Describe the solution you'd like**
Preload the following domains if the Adverts toggle is on for the service and if the Adverts enabled flag for the asset is set too true.

**Test**
```
https://gn-web-assets.api.bbc.com/ngas/latest/test/dotcom-bootstrap.js
https://gn-web-assets.api.bbc.com/ngas/latest/test/dotcom-ads.js
https://gn-web-assets.api.bbc.com/ngas/latest/test/dotcom-preroll.js
https://gn-web-assets.api.bbc.com/ngas/latest/test/dotcom-analytics.js
```

**Live**
```
https://gn-web-assets.api.bbc.com/ngas/latest/dotcom-bootstrap.js
https://gn-web-assets.api.bbc.com/ngas/latest/dotcom-ads.js
https://gn-web-assets.api.bbc.com/ngas/latest/dotcom-preroll.js
https://gn-web-assets.api.bbc.com/ngas/latest/dotcom-analytics.js
```

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves #8257

**Overall change:**
GNL have produced a set of un-versioned domains for some of their Advert logic that will enable us to preload them on pages that should serve ads.

**Code changes:**
Preload ad scripts on test / live environments

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh projec

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/Ad/Canonical/index.jsx [hunk 1]
```diff
@@ -19,22 +19,40 @@ const AdContainer = styled.section`
...
-    'https://gn-web-assets.api.bbc.com/ngas/dotcom-bootstrap-legacy.js';
+    'https://gn-web-assets.api.bbc.com/ngas/latest/dotcom-bootstrap-legacy.js';
   const useLiveSrc = isLive() || queryString.includes('ads-js-env=live');
   if (useLiveSrc) {
     return useLegacy ? adsLegacyLiveScript : adsLiveScript;
   }
   return useLegacy ? adsLegacyTestScript : adsTestScript;
 };
 
+const getPreloadUrls = (pageType, queryString) => {
```
Comment (karinathomasbbc) [inline src/app/containers/Ad/Canonical/index.jsx]: I can't see where this is used - do we need it?

Comment (chris-hinds) [reply src/app/containers/Ad/Canonical/index.jsx]: Used here https://github.com/bbc/simorgh/pull/8358/files#diff-2e2eb493e4304c784acc80b54089e86784f30e11d06cdb2e1b078c98a13eaaa1R92

File: src/app/containers/Ad/Canonical/index.jsx [hunk 2]
```diff
@@ -70,13 +92,20 @@ const CanonicalAd = ({ slotType, className }) => {
   return (
     <>
       <Helmet>
+        {/* PreLoad Ad Scripts */}
+        {getPreloadUrls(queryString).map(script => (
```
Comment (karinathomasbbc) [inline src/app/containers/Ad/Canonical/index.jsx:96]: @chris-hinds Should we only preload if ads enabled for service? Or is this logic already handled by the fact that we won't even show the Ads container if ads are not enabled?

Comment (chris-hinds) [reply src/app/containers/Ad/Canonical/index.jsx:96]: This logic should be handled by the container already

File: src/app/containers/Ad/Canonical/index.jsx [hunk 3]
```diff
@@ -19,20 +19,42 @@ const AdContainer = styled.section`
...
   const useLiveSrc = isLive() || queryString.includes('ads-js-env=live');
+
   if (useLiveSrc) {
     return useLegacy ? adsLegacyLiveScript : adsLiveScript;
   }
   return useLegacy ? adsLegacyTestScript : adsTestScript;
 };
 
+export const getPreloadUrls = queryString => {
+  const useLiveSrc = isLive() || queryString.includes('ads-js-env=live');
```
Comment (andrewscfc) [inline src/app/containers/Ad/Canonical/index.jsx]: This logic is duplicated here: https://github.com/bbc/simorgh/pull/8358/files#diff-2e2eb493e4304c784acc80b54089e86784f30e11d06cdb2e1b078c98a13eaaa1R30 

Should we just pass in `isLive` as a flag to `getBootstrapSrc`?

**Review by andrewscfc — APPROVED** [review]

**Review by joshcoventry — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
Can see an ad request being made and a place holder for the ad. Happy for this to be merged.

**Pull request merged**